controllers/datasciencecluster/datasciencecluster_controller.go

```diff
@@ -104,6 +105,26 @@ func (r *DataScienceClusterReconciler) Reconcile(ctx context.Context, req ctrl.R
 
 	instance := &instances.Items[0]
 
+	// If DSC CR exist and deletion CM exist
+	// delete DSC CR and let reconcile requeue
+	// sometimes with finalzier DSC CR wont get deleted, force to remove finalizer here
+	if upgrade.HasDeleteConfigMap(r.Client) {
+		if controllerutil.ContainsFinalizer(instance, finalizerName) {
+			if controllerutil.RemoveFinalizer(instance, finalizerName) {
```

Comment:
If there is a finalizer, won't it mean we has some undeleted ODH resource?

Reply:
normally when deletion happens, all child resource will be deleted first then the finalizer is removed from parent resource ( in this case the DSC CR). but i encountered several times, deletion of DSC CR hanging there and not get cleaned up. 
this happened in both cases of "directly uninstall operator from UI" and "directly delete DSC CR"
this is only to prevent the case, because of left finalizer to prevent DSC CR get removal and cause operator uninstallation hanging there